src/freenet/crypt/HMAC.java

```diff
@@ -53,6 +54,10 @@ public static boolean verify(HMAC hash, byte[] key, byte[] data, byte[] mac) {
     return MessageDigest.isEqual(mac, mac(hash, key, data));
   }
 
+  public static boolean verify(HMAC hash, byte[] key, byte[] data, byte[] mac, int length) {
```

Comment:
This isn't fool-proof; I wanted to get rid of the length parameter to avoid that type of problems.
IMHO we should enforce a length check on the key here ... and thow an IllegalArgumentException in case it's less than crypto-safe.